File: solutions/2825. Make String a Subsequence Using Cyclic Increments/2825.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  bool canMakeSubsequence(string_view str1, string_view str2) {
    int i = 0;

    for (const char c : str1)
      if (c == str2[i] || ('a' + ((c - 'a' + 1) % 26)) == str2[i])
        if (++i == str2.length())
          return true;

    return false;
  }
};

File: solutions/2825. Make String a Subsequence Using Cyclic Increments/2825.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def canMakeSubsequence(self, str1: str, str2: str) -> bool:
    i = 0

    for c in str1:
      if c == str2[i] or chr(
          ord('a') + (string.ascii_lowercase.index(c) + 1) % 26) == str2[i]:
        i += 1
        if i == len(str2):
          return True

    return False